h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -458,16 +458,24 @@
     return dout;
   }
 
-  @Override protected int fileHasHeader(byte[] bits, ParseSetup ps) {
+  @Override
+  protected int fileHasHeader(byte[] bits, ParseSetup parseSetup) {
     boolean hasHdr = true;
     String[] lines = getFirstLines(bits);
-    if (lines != null && lines.length > 0) {
-      String[] firstLine = determineTokens(lines[0], _setup._separator, _setup._single_quotes);
-      if (_setup._column_names != null) {
-        for (int i = 0; hasHdr && i < firstLine.length; ++i)
+    if (lines != null && lines.length > 1) {
```

Comment:
This doesn't seem correct to me. You will parse a single-line file that doesn't have a header as if it had a header.

Reply:
Right. This conditions has to stay at zero and do null checks afterwards.